apollo-router/src/plugins/telemetry/reload.rs

```diff
@@ -201,6 +202,11 @@ where
             return spanref.is_sampled();
         }
 
+        // always sample the router loading trace
+        if meta.name() == STARTING_SPAN_NAME {
```

Comment:
Suggest having an attribute called something like `router.lifecycle` or `sample/always` that can be attached to spans that we always want to sample rather than using span name. That way we can avoid nested spans being sampled.

Reply:
Ahh, although I guess if the root span is sampled then all children will be. So ignore this.

Reply:
yup, that's a nice property of this implem, the sampling decision is made only once (we basically had not other choice, because otherwise we would get incomplete traces)